### PR TITLE
The Great ESM Migration

### PR DESCRIPTION
**Pull request by jaketrent**
## Overture

Please see full commit messages for more context on the changes you see.

This is for ActionMenu only. Let's get one package right first. If you dig it, we'll expand to the rest of packagekind.

Once approved, I do not plan to merge but to do more commits and merge all the packages in one PR.

A couple other links re the file extensions (TS-specific) that might be helpful:

- https://github.com/microsoft/TypeScript/issues/16577#issuecomment-754941937
- https://github.com/microsoft/TypeScript/issues/16577#issuecomment-703190339

### CODE REVIEW

**Comment by jaketrent** [comment]
Note: Build is broken. Note avatar build:css busted. This is because the change to build makes it esm compatible only. Once we update all the packages to be esm, we should be set.

File: packages/actionmenu/src/react/arrow.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { HTMLPropsFor } from '@pluralsight/ps-design-system-util'
 import { css } from 'glamor'
 import React from 'react'
 
-import stylesheet from '../css'
+import stylesheet from '../css/index.js'
```
Comment (EdwardIrby) [inline packages/actionmenu/src/react/arrow.tsx:5]: @jaketrent did importing `.ts` break tree shaking?

Comment (jaketrent) [reply packages/actionmenu/src/react/arrow.tsx:5]: No, I don't think so. But I'm not sure why you ask, so maybe there's something I'm missing. Is there something you know about that we should verify related to tree shaking?

This change was made to provide explicit file extensions, required by esm, that TS allows but will not rewrite for us in compilation. (see linked TS issues in the PR)

Comment (EdwardIrby) [reply packages/actionmenu/src/react/arrow.tsx:5]: Didn't read those it's clear now. Thanks!

Comment (EdwardIrby) [reply packages/actionmenu/src/react/arrow.tsx:5]: However it looks like that is what's causing the test failures.  🤷🏿‍♂️

**Review by danethurber — APPROVED** [review]
approach seems simple to me. the js on imports seems weird to me - but if that's the requirement that let's do it. i could have expected tsc to add the js extensions 🤷

**Comment by jaketrent** [comment]
@danethurber Ditto, I would have thought tsc would do it. In the linked issues to the TS project, you can see the maintainers saying that rewriting the files like that is no in line with their project philosophy.

**Pull request closed without merging**